### PR TITLE
feat(component): add Badge props to Checkbox component

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

Add Badge props to Checkbox component

## Screenshots/Screen Recordings

Screenshot 2022-12-27 at 18 56 14

## Testing/Proof

Manual testing

### CODE REVIEW

**Review by eugene-polev — APPROVED** [review]

File: packages/docs/pages/checkbox.tsx
Comment (jorgemoya) [inline packages/docs/pages/checkbox.tsx:144]: Do you mean `specific information`? 

I also wonder if this fits well 🤔.  My take is that we can change this to `status information`.

File: packages/docs/pages/checkbox.tsx
Comment (jorgemoya) [inline packages/docs/pages/checkbox.tsx:168]: Lets demo a different label/variant for the disabled field, maybe use `warning/danger`.

File: packages/docs/pages/checkbox.tsx
Comment (jorgemoya) [inline packages/docs/pages/checkbox.tsx:144]:
```suggestion
                    specific information.
```

Missing a period.

File: packages/docs/pages/checkbox.tsx
Comment (bc-nick) [reply packages/docs/pages/checkbox.tsx]: @jorgemoya yeap, it was a typo. i mean `specific information`
But your version would be better
I have replaced with `status information`.

File: packages/docs/pages/checkbox.tsx
Comment (bc-nick) [reply packages/docs/pages/checkbox.tsx]: @jorgemoya updated the screenshot in description

File: packages/docs/pages/checkbox.tsx
Comment (bc-nick) [reply packages/docs/pages/checkbox.tsx]: Thanks. I have fixed

**Review by jorgemoya — APPROVED** [review]
Lets see if the @bigcommerce/dev-docs-team can 👀 before we merge.

**Comment by bc-traciporter** [comment]
@jorgemoya, @bc-andreadao is on vacation this week, can you wait until after the new year when she returns to review and approve this PR?

**Comment by bc-nick** [comment]
> @jorgemoya, @bc-andreadao is on vacation this week, can you wait until after the new year when she returns to review and approve this PR?

@bc-traciporter need to close the current task this week and this pull request would be better closed this week. is there any possibility to merge it this week and release it?

**Comment by bc-traciporter** [comment]
@jorgemoya sure we can help out.  Does she perform just an editorial review or do we need to test anything?

**Comment by jorgemoya** [comment]
> @bc-traciporter need to close the current task this week and this pull request would be better closed this week. is there any possibility to merge it this week and release it?

I don't see us releasing this week tbh. We have some pending PRs and people on PTO. Lets wait for next week.

> @jorgemoya sure we can help out. Does she perform just an editorial review or do we need to test anything?

Just an editorial review for the new documentation tidbits. Thanks!

File: packages/big-design/src/components/Checkbox/Checkbox.tsx
Comment (chanceaclark) [inline packages/big-design/src/components/Checkbox/Checkbox.tsx:78]: Do you think we'd want the badge to be read along with the label for accessibility 🤔 Having it outside the `label` will remove the badge text from being read along with the label and would require additional keyboard navigation to read it.
```suggestion
          <CheckboxLabel
            aria-hidden={disabled}
            disabled={disabled}
            hidden={hiddenLabel}
            htmlFor={id}
            id={labelId}
          >
            {label}
            {badge ? <Badge marginLeft="xSmall" {...badge} /> : null}
          </CheckboxLabel>
```

**Review by chanceaclark — COMMENTED** [review]
I have a slight accessibility comment for you, otherwise looking good. Would you also mind adding a quick test for this too?

File: packages/docs/pages/checkbox.tsx
Comment (bc-andreadao) [inline packages/docs/pages/checkbox.tsx:143]:
```suggestion
                    <Code primary>Checkboxes</Code> support <Code primary>Badge</Code> that display{' '}
```

File: packages/big-design/src/components/Checkbox/Checkbox.tsx
Comment (bc-nick) [reply packages/big-design/src/components/Checkbox/Checkbox.tsx]: @chanceaclark thanks for this point. I updated and added a small test

File: packages/big-design/src/components/Checkbox/spec.tsx
Comment (chanceaclark) [inline packages/big-design/src/components/Checkbox/spec.tsx:63]: 🍹 A snapshot test might be a little over the top here 🤔 Maybe a `expect(getByText('Badge')).toBeInTheDocument()` could be sufficient.

File: packages/big-design/src/components/Checkbox/Checkbox.tsx
Comment (chanceaclark) [inline packages/big-design/src/components/Checkbox/Checkbox.tsx:67]: 🍹 Useless fragment

**Review by chanceaclark — APPROVED** [review]

**Pull request merged**
